### PR TITLE
miami CParticleObject

### PR DESCRIPTION
**Pull request by Fire-Head**

### CODE REVIEW

File: src/objects/ParticleObject.cpp [hunk 1]
```diff
@@ -162,226 +162,240 @@ CParticleObject::AddObject(uint16 type, CVector const &pos, CVector const &targe
...
+{
+	CParticleObject *pobj = pUnusedListHead;
+	
+	ASSERT(pobj != NULL);
+	
+	if ( pobj == NULL )
+	{
+		printf("Error: No particle objects available!\n");
+		return NULL;
+	}
```
Comment (ShFil119) [inline src/objects/ParticleObject.cpp:368]: nil

File: src/objects/ParticleObject.cpp [hunk 2]
```diff
@@ -979,63 +877,107 @@ void CParticleObject::UpdateClose(void)
...
+							splashpos = pos + CVector(0.01f*-fCos, 0.01f*-fSin, 0.0f);
+							splashvel = vel + CVector(0.0f, 0.0f, CGeneral::GetRandomNumberInRange(0.004f, 0.008f));
+							
+							CParticle::AddParticle(PARTICLE_CAR_SPLASH, splashpos, splashvel, NULL,
+								CGeneral::GetRandomNumberInRange(0.005f, 0.0075f), this->m_Color, 0, 0, 1, 300);
+						}
+						for ( int32 i = 0; i < this->m_nNumEffectCycles; i++ )
+						{
+							CParticle::AddParticle(this->m_ParticleType, pos, vel, NULL, 0.0f, this->m_Color);
+						}
```
Comment (ShFil119) [inline src/objects/ParticleObject.cpp:923]: also nil

File: src/objects/ParticleObject.cpp [hunk 3]
```diff
@@ -979,63 +877,107 @@ void CParticleObject::UpdateClose(void)
...
+							
+						CParticle::AddParticle(PARTICLE_SPLASH, splashpos, splashvel, NULL,
+							CGeneral::GetRandomNumberInRange(0.001f, 0.005f), this->m_Color, 0, 0, 1, 1000);
+							
+						splashpos = pos + CVector(0.015f*-fCos, 0.015f*-fSin, 0.0f);
+						splashvel = vel + CVector(0.015f*-fCos, 0.015f*-fSin, CGeneral::GetRandomNumberInRange(0.004f, 0.008f));
+							
+						CParticle::AddParticle(PARTICLE_SPLASH, splashpos, splashvel, NULL,
+							CGeneral::GetRandomNumberInRange(0.001f, 0.005f), this->m_Color, 0, 0, 1, 1000);
+					}
```
Comment (ShFil119) [inline src/objects/ParticleObject.cpp:966]: etc

File: src/objects/ParticleObject.cpp [hunk 4]
```diff
@@ -979,63 +877,107 @@ void CParticleObject::UpdateClose(void)
...
-					
-					flashpos += velstep;
-					CParticle::AddParticle(PARTICLE_GUNFLASH, flashpos, CVector(0.0f, 0.0f, 0.0f), NULL, 0.2f*size, flashcolor);
-					
+					CVector vel = this->m_vecTarget;
 					
-					CParticle::AddParticle(PARTICLE_GUNFLASH, pos, CVector(0.0f, 0.0f, 0.0f), NULL, 0.0f, flashcolor);
+					for ( int32 i = 0; i < 3; i++ )
+					{
+						CParticle::AddParticle(PARTICLE_CAR_SPLASH, pos, vel, NULL, 0.001f, this->m_Color, 0, 0, 1, 1000);
```
Comment (ShFil119) [inline src/objects/ParticleObject.cpp:978]: +1

**Comment by erorcun** [comment]
lgtm

**Pull request merged**